src/lib/deterministic-fsm/deterministic-state-machine.ts

```diff
@@ -0,0 +1,13 @@
+import { DeterministicTransitions, deterministicTransitionFn } from './deterministic-transition.fn';
+import { StateMachine, MachineNode } from '../state-machine-api';
+import { GenericStateMachine } from '../state-machine';
+
+export const deterministicStateMachine: (
+  initialState: MachineNode,
+  finalNodes: Set<MachineNode>,
+  transitions: DeterministicTransitions,
+) => StateMachine<void> = (
+  initialState: MachineNode,
+  finalNodes: Set<MachineNode>,
+  transitions: DeterministicTransitions,
+) => new GenericStateMachine<void>(initialState, finalNodes, deterministicTransitionFn(transitions));
```

Comment:
why `DeterministicStateMachine` is not a class that **extend from GenericStateMachine**?
I mean, sound something weir read this

```typescript
 const machine = deterministicStateMachine(initialState, finalNodes, transitions);
```

If we read the expression we can't imagine that we have an instance from a GenericStateMachine, because this is happening behind the shadows, we could understood what is happening if we read this expression

```typescript
 const machine = new DeterministicStateMachine(initialState, finalNodes, transitions);
```

Reply:
I guess is because the deterministic functionality is given by the transitions, not by the machine itself. The transitions are the one being deterministic.

Reply:
ummm yep I now that transitions is like the decorator in this case but, probably we are not using exactly the decorator pattern.

Probably the problem could be the naming, I mean, like `deterministicStateMachine` is not a constructor itself, why we could not call this method `getDeterministicStateMachine`? I don't know if its good idea. :thinking: